src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc

```diff
@@ -0,0 +1,218 @@
+}
+
+void SDLPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  using namespace application_manager;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& response = event.smart_object();
+  const uint32_t corr_id = event.smart_object_correlation_id();
+
+  LOG4CXX_TRACE(logger_,
+                "Received event with function id: "
+                    << event.id() << " and correlation id: " << corr_id);
+
+  smart_objects::SmartObject pending_request;
+  if (pending_requests_.find(corr_id) == pending_requests_.end()) {
+    LOG4CXX_DEBUG(logger_, "corr id " << corr_id << " NOT found");
+    return;
+  }
+  pending_request = pending_requests_[corr_id];
+  pending_requests_.erase(corr_id);
+  if (app_ids_.empty()) {
+    LOG4CXX_ERROR(logger_, "app_ids is empty");
+    return;
+  }
+  uint32_t app_id = app_ids_.front();
+  app_ids_.pop();
+  auto app = application_manager_.application(app_id);
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application not found " << app_id);
+    return;
+  }
+
+  if (resumption::IsResponseSuccessful(response)) {
+    LOG4CXX_DEBUG(logger_, "Resumption of subscriptions is successful");
+
+    application_manager_.SubscribeAppForWayPoints(app);
+
+    for (auto& freezed_resumption : freezed_resumptions_) {
+      auto corr_id = freezed_resumption.request_to_send_
+                         .message[strings::params][strings::correlation_id]
+                         .asInt();
+      RaiseFakeSuccessfulResponse(response, corr_id);
+      application_manager_.SubscribeAppForWayPoints(freezed_resumption.app_id);
+    }
+    freezed_resumptions_.clear();
+  } else {
+    LOG4CXX_DEBUG(logger_, "Resumption of subscriptions is NOT successful");
+
+    if (freezed_resumptions_.empty()) {
+      LOG4CXX_DEBUG(logger_, "freezed resumptions is empty");
+      return;
+    }
+
+    ResumptionAwaitingHandling freezed_resumption = freezed_resumptions_.back();
+    freezed_resumptions_.pop_back();
+    auto resumption_req = freezed_resumption.request_to_send_;
+    const uint32_t cid =
+        resumption_req.message[strings::params][strings::correlation_id]
+            .asInt();
+    const hmi_apis::FunctionID::eType fid =
+        static_cast<hmi_apis::FunctionID::eType>(
+            resumption_req.message[strings::params][strings::function_id]
+                .asInt());
+    subscribe_on_event(fid, cid);
+    auto request =
+        std::make_shared<smart_objects::SmartObject>(resumption_req.message);
+    LOG4CXX_DEBUG(logger_,
+                  "Subscribing for event with function id: "
+                      << fid << " correlation id: " << cid);
+    pending_requests_[cid] = *request;
+    LOG4CXX_DEBUG(logger_,
+                  "Sending request with fid: " << fid << " and cid: " << cid);
+    application_manager_.GetRPCService().ManageHMICommand(request);
+  }
+}
+
+void SDLPendingResumptionHandler::HandleResumptionSubscriptionRequest(
```

Comment:
Move this function before ResumptionVevert for redability

Reply:
Done in 25b2f2299d5caa7432c0e51f6208f3696143dc85